meta-balena-common/recipes-core/systemd/timeinit/timeinit-rtc.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+#
+# Copyright 2020 Balena Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+. /usr/libexec/os-helpers-logging
+. /usr/libexec/os-helpers-time
+
+RTC_DEVICE=/dev/rtc
+
+# Don't bother updating or reporting errors for small differences.
+TIME_DIFF_THRESHOLD=2
```

Comment:
This was mostly to prevent the clock being set unnecessarily when the time is actually correct but the seconds have rolled-over between reading the system time and reading the RTC time.

I thought that it made sense to make this flexible and to not bother updating for very small differences which will subsequently be corrected by `chrony` in most cases. Arguably this could be set to `1`, but I thought a value of `2` was probably a good enough trade-off between accuracy and updating unnecessarily.